miner/worker.go

```diff
@@ -1164,6 +1175,17 @@ loop:
 			// inconsistent state, so we cannot pack more transactions.
 			w.eth.TxPool().RemoveTx(tx.Hash(), true)
 			circuitCapacityReached = true
+			txNumInCcc, err := w.circuitCapacityChecker.GetTxNum()
```

Comment:
I think we should merge these two `case` branches into one to reduce code duplication. (Ping me if you need me to do this)

Reply:
Add a function [checkCurrentTxNum](https://github.com/scroll-tech/go-ethereum/pull/505/commits/6387d3f2b22e0928ff7ab62e53742ab687457747#diff-689874b30f6f905ce6c47cdefeddcf052b467a7936a981f4b6cf38939e10d924R1213) in commit [6387d3f](https://github.com/scroll-tech/go-ethereum/pull/505/commits/6387d3f2b22e0928ff7ab62e53742ab687457747).